lesson-demo/src/main/java/com/bobocode/pool/ConnectionProxy.java

```diff
@@ -0,0 +1,307 @@
+package com.bobocode.pool;
+
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.SQLClientInfoException;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Queue;
+import java.util.concurrent.Executor;
+
+public class ConnectionProxy implements Connection {
+
+    // todo: 1. store a physical connection and a pool reference
+    // todo: 2. override method close
+    // todo: 3. delegate all method invocations to a physical connection
+
+    private Connection connection;
```

Comment:
@NikolayProkopchuk these two fields should never change, hence they should be `final`

Reply:
fixed, thank you:)